addons/test_mail/tests/test_mail_composer.py

```diff
@@ -1852,6 +1859,137 @@ def test_mail_composer_delete(self):
+                [[[], diff_recipients]]):
+            with self.subTest(template_changes=template_changes, records_changes=records_changes):
+                test_template_values = dict(base_template_values)
+                for change in template_changes:
+                    test_template_values.update(change)
+                self.template.write(test_template_values)
+                for record_n, record in enumerate([self.test_records[0], self.test_records[1]]):
+                    record.write({**base_record_values[record_n], **records_changes[record_n]})
+                composer = _instanciate_composer()
+                with self.mock_mail_gateway(mail_unlink_sent=False), self.mock_mail_app():
+                    composer._action_send_mail()
+                self.assertEqual(len(self._new_mails), 2, f'''Should have sent all email.
+                                 template values: {test_template_values}
+                                 records values: {records_changes}''')
+                for record in [self.test_records[0], self.test_records[1]]:
+                    expected_subject = base_template_values['subject'] if test_template_values['subject'] == base_template_values['subject'] else record.name
+                    expected_body = base_template_values['body_html'] if test_template_values['body_html'] == base_template_values['body_html'] else f'<p>{record.name}</p>'
+                    expected_attachment_info = []
+                    if test_template_values['attachment_ids'] and test_template_values['attachment_ids'][0] != Command.clear():
+                        expected_attachment_info.append({'name': 'SomeCommonAttachment.txt', 'type': 'text/plain'})
+                    expected_test_reports = test_template_values['report_template_ids']
+                    if expected_test_reports and expected_test_reports[0] != Command.clear():
+                        expected_attachment_info.append(
+                            {'name': f'{"TestReport2" if expected_test_reports[0] == self.test_report_2.id else "TestReport3"} for {record.name}.html',
+                             'type': 'text/plain'},
+                        )
+                    self.assertMailMailWEmails(
+                        [email_normalize(record.email_from)],
+                        'sent',
+                        author=self.env.user.partner_id,
+                        content=expected_body,
+                        mail_message=record.message_ids[0],
+                        email_values={
+                            'attachments_info': expected_attachment_info,
+                            'body': expected_body,
+                            'email_from': record.user_id.email_formatted,
+                            'subject': expected_subject,
+                        },
+                    )
+
+        # expect duplicates
+        for with_composer_attachment, template_changes in zip([False, True], [[], [same_attachments]] * 2):
```

Comment:
Don't understand the "with composer attachment", is it replacing template attachments / adding attachments ? Have to say it is worth a small comment ^^